### PR TITLE
Recursive directory scan and result summary fix

### PR DESCRIPTION
**Pull request by penenadpi**
- added support for recursive subdirectory traversal within IaC archive
- fixed the outputs of several tools (checkstyle, hadolint, htmlhint, ts-lint, es-lint)
- fixed the bug related to additional extensions within Terraform and Tosca archives
- covering the scenario for unsupported file types within IaC archive

### CODE REVIEW

**Review by cankarm — APPROVED** [review]
For me it seems OK for now. It would be nice to merge it quick, so we have the fixes covered for the users.

File: src/iac_scan_runner/compatibility.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ class Compatibility:
         "js": ["es-lint", "ts-lint"],
         "html": ["htmlhint"],
         "docker": ["hadolint"],
+        "other": [""],
```
Comment (anzoman) [inline src/iac_scan_runner/compatibility.py:16]: Why list with empty string and not just empty list?

File: src/iac_scan_runner/compatibility.py [hunk 2]
```diff
@@ -44,43 +45,64 @@ def check_iac_type(self, iac_directory: str) -> List[str]:
         scanned_yaml = []
         scanned_java = []
         scanned_html = []
-
+        scanned_js = []
+        scanned_docker = []
+        scanned_other = []
```
Comment (anzoman) [inline src/iac_scan_runner/compatibility.py:50]: Do we really need `scanned_<something>`?
I suggest extracting this into a separate object. I see you already have `self.scanned_files`, so you can append directly into that.

File: src/iac_scan_runner/compatibility.py [hunk 3]
```diff
@@ -44,43 +45,64 @@ def check_iac_type(self, iac_directory: str) -> List[str]:
...
             self.scanned_files["shell"] = str(scanned_shell)
             self.scanned_files["yaml"] = str(scanned_yaml)
             self.scanned_files["java"] = str(scanned_java)
             self.scanned_files["html"] = str(scanned_html)
+            self.scanned_files["js"] = str(scanned_js)
+            self.scanned_files["docker"] = str(scanned_docker)
+            self.scanned_files["other"] = str(scanned_other)
+                                                
+            types = set(types)
+            print(types)
```
Comment (anzoman) [inline src/iac_scan_runner/compatibility.py:104]: This should probably be removed.

File: src/iac_scan_runner/results_summary.py [hunk 4]
```diff
@@ -70,6 +70,59 @@ def summarize_outcome(
                 self.outcomes[check]["status"] = "Problems"
                 return "Problems"
 
+        if check == "htmlhint":
+            if outcome.find("no errors")>-1:
+                self.outcomes[check]["status"] = "Passed"
+                return "Passed"
```
Comment (anzoman) [inline src/iac_scan_runner/results_summary.py:76]: Add a TODO to extract "Passed" and "Problems" into an `Enum` object and use them.

File: src/iac_scan_runner/scan_runner.py [hunk 5]
```diff
@@ -142,49 +144,42 @@ def _run_checks(self, selected_checks: Optional[List], scan_response_type: ScanR
...
-                            scan_output[selected_check] = check_output.to_dict()
-                        else:
-                            # TODO: Discuss the format of this output
-                            scan_output += f"### {selected_check} ###\n{check_output.to_string()}\n\n"
-
+                        scan_output[selected_check] = check_output.to_dict()                        
                         write_string_to_file(check.name, dir_name, scan_output[check.name]["output"])
                         self.results_summary.summarize_outcome(selected_check, scan_output[check.name]["output"], self.compatibility_matrix.scanned_files, Compatibility.compatibility_matrix)
-         
+
```
Comment (anzoman) [inline src/iac_scan_runner/scan_runner.py:158]: Remove these new empty spaces.

File: src/iac_scan_runner/scan_runner.py [hunk 6]
```diff
@@ -142,49 +144,42 @@ def _run_checks(self, selected_checks: Optional[List], scan_response_type: ScanR
...
+                        
                     else:
                         non_compatible_checks.append(check.name)
                         write_string_to_file(check.name, dir_name, "No files to scan")
                         self.results_summary.summarize_no_files(check.name)
 
             self.results_summary.dump_outcomes(random_uuid)
             self.results_summary.generate_html_prioritized(random_uuid)
-
+
```
Comment (anzoman) [inline src/iac_scan_runner/scan_runner.py:166]: Remove these new empty space.

File: src/iac_scan_runner/utils.py [hunk 7]
```diff
@@ -97,3 +97,18 @@ def write_html_to_file(file_name: str, output_value: str):
             text_file.write(output_value)
     except Exception as e:
         raise Exception(f"Error storing HTML to file: {str(e)}.")           
+
+
+def file_to_string(file_path: str):
```
Comment (anzoman) [inline src/iac_scan_runner/utils.py:102]: Missing return type.

```suggestion
def file_to_string(file_path: str) -> str:
```

File: src/iac_scan_runner/utils.py [hunk 8]
```diff
@@ -97,3 +97,18 @@ def write_html_to_file(file_name: str, output_value: str):
...
+    """
+    Reads the file given by path and returns its contents as string output
+    :param file_path: Path of the file
+    :return output: Content read from file in form of a string
+    """
+    output = ""
+    try:
+        with open(file_path, "r") as text_file:
+            output = str(text_file.read())
+    except Exception as e:
```
Comment (anzoman) [inline src/iac_scan_runner/utils.py:112]: This exception is probably too broad, try to narrow it down.

**Review by anzoman — CHANGES_REQUESTED** [review]
Thanks @penenadpi, here is my review.

File: src/iac_scan_runner/compatibility.py (hunk 1 above)
Comment (penenadpi) [reply src/iac_scan_runner/compatibility.py:16]: Ok, gonna fix this one

File: src/iac_scan_runner/compatibility.py (hunk 3 above)
Comment (penenadpi) [reply src/iac_scan_runner/compatibility.py:104]: I agree, this one remained for debugging. Maybe we should include the adoption of loguru or something similar for next version, could help out a bit.

File: src/iac_scan_runner/results_summary.py (hunk 4 above)
Comment (penenadpi) [reply src/iac_scan_runner/results_summary.py:76]: I agree, should be added

File: src/iac_scan_runner/utils.py (hunk 7 above)
Comment (penenadpi) [reply src/iac_scan_runner/utils.py:102]: I definitely forgot to add it, gonna fix it

File: src/iac_scan_runner/scan_runner.py (hunk 6 above)
Comment (penenadpi) [reply src/iac_scan_runner/scan_runner.py:166]: Ok

File: src/iac_scan_runner/scan_runner.py (hunk 5 above)
Comment (penenadpi) [reply src/iac_scan_runner/scan_runner.py:158]: Removed

File: src/iac_scan_runner/compatibility.py (hunk 2 above)
Comment (penenadpi) [reply src/iac_scan_runner/compatibility.py:50]: Would work on that later, as code would need to be quite refactored

**Review by anzoman — APPROVED** [review]
@penenadpi thanks for your contribution, let's merge this one!

**Pull request merged**